docs/source/getting-started/tabular-data/data-prep.ipynb

```diff
@@ -0,0 +1,279 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Data Preparation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This section explains the limitations of the `with_columns` and `filter` methods in our `dp` plugin for Polars. \n",
+    "\n",
+    "Although these methods function similarly to normal Polars versions, in this context there are privacy implications that may not be immediately apparent. \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set Up"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import polars as pl \n",
+    "import opendp.prelude as dp\n",
+    "import seaborn as sns \n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "dp.enable_features(\"contrib\")\n",
+    "sns.set_theme(style='darkgrid')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
+      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+      "  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n",
+      "  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n",
+      "100 5933k  100 5933k    0     0  1088k      0  0:00:05  0:00:05 --:--:-- 1139k\n",
+      "Archive:  data.zip\n",
+      "  inflating: sample_FR_LFS.csv       \n",
+      "  inflating: __MACOSX/._sample_FR_LFS.csv  \n"
+     ]
+    }
+   ],
+   "source": [
+    "!curl 'https://github.com/opendp/dp-test-datasets/blob/master/data/sample_FR_LFS.csv.zip?raw=true' --location --output data.zip\n",
+    "!unzip data.zip\n",
+    "df = pl.scan_csv(\"sample_FR_LFS.csv\", ignore_errors=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "estimated_max_partition_len = 60_000_000\n",
+    "\n",
+    "context = dp.Context.compositor(\n",
+    "    data=df,\n",
+    "    privacy_unit=dp.unit_of(contributions=36),\n",
+    "    privacy_loss=dp.loss_of(epsilon=1.0),\n",
+    "    split_evenly_over=5,\n",
+    "    margins={\n",
+    "        (\"YEAR\", ): dp.Margin(max_partition_length=estimated_max_partition_len, max_partition_contributions=4),\n",
+    "        (\"ILOSTAT\", ): dp.Margin(max_partition_length=estimated_max_partition_len),\n",
+    "        (): dp.Margin(public_info= \"lengths\",max_partition_length=estimated_max_partition_len, max_num_partitions=1),\n",
+    "    },\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Limitations of `with_columns`"
```

Comment:
Why are these "limitations"? I may be misinterpreting something, but are these things drawbacks?

Reply:
not necessarily drawbacks, but basically that these methods might not have the same implications as when they are used in polars. maybe mike has a better word for it?